src/server/endpoints/covidcast.py

```diff
@@ -140,18 +185,20 @@ def handle():
     fields_int = ["time_value", "direction", "issue", "lag", "missing_value", "missing_stderr", "missing_sample_size"]
     fields_float = ["value", "stderr", "sample_size"]
     is_compatibility = is_compatibility_mode()
+
     if is_compatibility:
-        q.set_order("signal", "time_value", "geo_value", "issue")
+        original_source = source_signal_pairs[0].source
```

Comment:
an alternative approach is to also return query the source field in the fields but the filter out the fields you don't want in the compatibility mode before yielding it